### PR TITLE
NEWS.md update for sparklyr 1.3 release

### PR DESCRIPTION
**Pull request by yl790**
This is just a draft with features and bug fixes implemented so far.
Don't worry, there is still plenty of time to be part of the spectacular success that is sparklyr 1.3.

### CODE REVIEW

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2550-1/ee6f57a5-7578-478a-bb30-f919b0a864b2.txt).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2550-2/025819da-97fa-46ec-87b2-9238aaf4583d.txt).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2550-3/e64a6c7b-bb3e-406e-8384-8368c74a5b50.txt).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2550-4/86476cc5-6c27-41f4-bc38-df87b16720d6.txt).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2550-5/73875085-80ec-4565-99dc-40eb7eed56ab.txt).

File: NEWS.md [hunk 1]
```diff
@@ -1,3 +1,72 @@
...
+  enabled Spark connections
+
+- For arrow version 0.17 or above, sparklyr will ensure POSIXt columns have timezone
+  set before being serialized by arrow (columns without timezone will assume
+  `Sys.timezone()`)
+
+- Embedded nul bytes are removed from strings when reading strings from Spark to R
+  (#2250)
+
+- `spark_read()` method is implemented to allow user-defined R functions to be run
```
Comment (falaki) [inline NEWS.md:45]:
```suggestion
- New `spark_read()` method to allow user-defined R functions to be run
```

File: NEWS.md [hunk 2]
```diff
@@ -1,3 +1,72 @@
...
+  on Spark workers to import data into a Spark data frame
+
+- `spark_write()` method is implemented allow user-defined functions to be run on
```
Comment (falaki) [inline NEWS.md:48]:
```suggestion
- New `spark_write()` method to allow user-defined functions to be run on
```

File: NEWS.md [hunk 3]
```diff
@@ -1,3 +1,72 @@
...
+
+- Avro functionalities such as `spark_read_avro()`, `spark_write_avro()`,
+  `sdf_from_avro()`, and `sdf_to_avro()` are implemented and can be optionally
+  enabled with `spark_connect(..., package = "avro")`
+
+### Misc
+
+- Fixed warnings for deprecated functions (#2431)
+
+- More test cases enabled for Databricks Connect
```
Comment (falaki) [inline NEWS.md:59]:
```suggestion
- More test coverage for Databricks Connect and Databricks Notebook modes
```

File: NEWS.md [hunk 4]
```diff
@@ -1,3 +1,72 @@
...
+- Enabled transform function of `spark_apply()` to return nested lists
+
+- Added option to return R objects instead of Spark data frame rows from transform
+  function of `spark_apply`
+
+- `sdf_collect()` now supports fetching Spark data frame row-by-row rather than
+  column-by-column, and fetching rows using iterator instead of collecting all
+  rows into memory
+
+### Connection
```
Comment (falaki) [inline NEWS.md:24]:
```suggestion
### Connections
```

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2550-6/554adf90-d5e8-4cb3-936e-0d70b5247e6e.txt).

File: NEWS.md [hunk 5]
```diff
@@ -1,3 +1,72 @@
...
+
+### Spark ML
+
+- `ml_evaluate()` methods are implemented for ML clustering and classification models
+
+### Distributed R
+
+- Created helper methods to integrate Spark SQL higher-order functions with
+  `dplyr::mutate`
+
```
Comment (javierluraschi) [inline NEWS.md:11]: I think this one fits better under `Data`, so removing form this section and suggesting another edit bellow:

```suggestion
- Created helper methods to integrate Spark SQL higher-order functions with
  `dplyr::mutate`

```

File: NEWS.md [hunk 6]
```diff
@@ -1,3 +1,72 @@
...
+- New `spark_read()` method to allow user-defined R functions to be run
+  on Spark workers to import data into a Spark data frame
+
+- `spark_write()` method is implemented allow user-defined functions to be run on
+  Spark workers to export data from a Spark data frame
+
+- Avro functionalities such as `spark_read_avro()`, `spark_write_avro()`,
+  `sdf_from_avro()`, and `sdf_to_avro()` are implemented and can be optionally
+  enabled with `spark_connect(..., package = "avro")`
+
```
Comment (javierluraschi) [inline NEWS.md:54]: I think these ones fit better into `Data`, as in "Data Import/Export/Transformation"

```suggestion
### Data

- Created helper methods to integrate Spark SQL higher-order functions with
  `dplyr::mutate`
  
- New `spark_read()` method to allow user-defined R functions to be run
  on Spark workers to import data into a Spark data frame

- `spark_write()` method is implemented allow user-defined functions to be run on
  Spark workers to export data from a Spark data frame

- Avro functionalities such as `spark_read_avro()`, `spark_write_avro()`,
  `sdf_from_avro()`, and `sdf_to_avro()` are implemented and can be optionally
  enabled with `spark_connect(..., package = "avro")`

```

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2550-8/e4ed6004-4851-4682-aed7-1082ba1d2678.txt).

**Comment by falaki** [comment]
Databricks Connect tests failed. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2550-7/261ae268-982b-486a-a4af-7fbaa37b6766.txt).

**Comment by falaki** [comment]
Databricks Connect tests failed. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2550-9/e82d3ba4-4c9e-47a8-a8f8-0f966f335e3f.txt).

**Comment by falaki** [comment]
Databricks Connect tests failed. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2550-10/0bd180ab-b7f8-4a29-af7b-011bdb0903c3.txt).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2550-11/9ecafa85-ca92-4975-97e2-fe0c1e09b7fc.txt).

**Comment by yl790** [comment]
@nealrichardson Many thanks for improving Arrow functionalities in sparklyr!
In this pull request I mentioned recent Arrow-related changes under the `Serialization` section of NEWS.md for the upcoming sparklyr 1.3 release.
Please let me know if anything is missing.

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2550-12/560b5cf5-8912-4623-86f9-39d2473390dc.txt).

File: NEWS.md [hunk 7]
```diff
@@ -1,3 +1,83 @@
...
+- Hive integration can now be disabled by configuration in `spark_connect()` (#2465)
+
+- A JVM object reference counting bug affecting secondary Spark connections was fixed
+  (#2515)
+
+- Revised JObj envs initialization for Databricks connections (#2533)
+
+### Serialization
+
+- The new `arrow::write_to_raw()` functionality will be used if available in arrow-
```
Comment (nealrichardson) [inline NEWS.md:35]: This isn't really interesting to a user of `sparklyr`. It's essentially a refactoring of the `arrow` code with no change in functionality.

File: NEWS.md [hunk 8]
```diff
@@ -1,3 +1,83 @@
...
+  enabled Spark connections
+
+- For arrow version 0.17 or above, sparklyr will ensure POSIXt columns have timezone
```
Comment (nealrichardson) [inline NEWS.md:38]: I think the feature here is that timezones, if present in the data, are correctly represented now. I don't think more needs to be said.

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2550-13/9a75e301-fc8f-4a58-843b-f6789f9c3de8.txt).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2550-14/64aea5e4-102a-4308-b435-0cfc990e83a0.txt).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2550-17/07d6704b-5217-4858-a417-b26b3fd38d91.txt).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2550-18/96a9d062-845b-4fd5-89fb-3d1e114f056e.txt).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2550-19/f38e9ecd-890a-4a07-84ef-c663c7a034b0.txt).

**Pull request merged**